Set single primary font (Patron) (#1306)
redpanda-data/documentation: modules/reference/pages/rpk/rpk-x-options.adoc

[TIP]
====
* For persistent configuration across commands and sessions, we recommend using xref:get-started:config-rpk-profile.adoc[rpk profiles] instead of environment variables or `-X` flags.
* `rpk` supports command-line (tab) completion for `-X` flag keys.
* Each `rpk` command's `-help` text prints information specific to the command. To view a description of `-X` options, run `rpk -X list` to list supported options, or run `rpk -X help` to get details about supported options.
====

== Configuration priority

`rpk` resolves configuration values in the following priority order (highest to lowest):

1. **Command-line flags** (including `-X` options): Highest priority, applies to current command only
2. **Environment variables**: Medium priority, lasts for shell session
   - **RPK_*** variables (modern): Higher precedence

>    - **REDPANDA_*** variables (legacy): Lower precedence

Reviewer: ditto

3. **rpk profile** (`rpk.yaml`): Lower priority, persistent across sessions (recommended)
4. **Redpanda configuration** (`redpanda.yaml` rpk section): Lowest priority, system-wide defaults

NOTE: If both `RPK_*` and `REDPANDA_*` environment variables are set for the same option, the `RPK_*` variable takes precedence.

== Environment variables

=== RPK_* environment variables

> Every `-X` option has a corresponding `RPK_*` environment variable. Convert by prefixing with `RPK_` and replacing dots with underscores:

Reviewer: ```suggestion
Every `-X` option has a corresponding `RPK_*` environment variable. Convert by prefixing with `RPK_` and replacing dots with underscores, for example:
```

Or something like that, because these are not all the environment variables.

Reviewer 2: now we have all env variables hehe https://github.com/redpanda-data/docs/pull/1306/commits/9d19e2da56625fcb648e04d590264af6763e911d 

(thanks for the list btw)

[cols="1m,1m"]
|===
|`-X` Option |Environment Variable

|brokers |RPK_BROKERS
|tls.enabled |RPK_TLS_ENABLED
|tls.ca |RPK_TLS_CA
|user |RPK_USER
|pass |RPK_PASS
|sasl.mechanism |RPK_SASL_MECHANISM
|admin.hosts |RPK_ADMIN_HOSTS
|===

> === Legacy REDPANDA_* environment variables

Reviewer: ditto

[TIP]
====

> * For persistent configuration across commands and sessions, we recommend using xref:get-started:config-rpk-profile.adoc[rpk profiles] instead of environment variables or `-X` flags.

Reviewer 3: ```suggestion
* For persistent configuration across commands and sessions, Redpanda Data recommends using xref:get-started:config-rpk-profile.adoc[rpk profiles] instead of environment variables or `-X` flags.
```
Style guide

Reviewer 2: edited to Redpanda Data

* `rpk` supports command-line (tab) completion for `-X` flag keys.
* Each `rpk` command's `-help` text prints information specific to the command. To view a description of `-X` options, run `rpk -X list` to list supported options, or run `rpk -X help` to get details about supported options.
====

== Configuration priority

`rpk` resolves configuration values in the following priority order (highest to lowest):

> 1. **Command-line flags** (including `-X` options): Highest priority, applies to current command only

Reviewer 3: I think we could probably get away with leaving out "Highest priority", "Medium," etc from these descriptions, to make these more succinct, since we already indicate in the preceding line that it's highest to lowest.

Reviewer 2: I was on the fence for this one, but I think the numbers already make this quite obvious. I'll remove it.

> 2. **Environment variables**: Medium priority, `RPK_*` env variables lasts for shell session

Reviewer 3: ```suggestion
2. **Environment variables**: Medium priority, `RPK_*` environment variables lasts for shell session
```
I'm not sure if "env var/variable" is standard in our docs, so just in case

3. **rpk profile** (`rpk.yaml`): Lower priority, persistent across sessions (recommended)

> 4. **Redpanda configuration** (`redpanda.yaml` rpk section): Lowest priority, system-wide defaults

Reviewer 3: Is this what's referred to in the [first line](https://deploy-preview-1306--redpanda-docs-preview.netlify.app/current/reference/rpk/rpk-x-options/) of this doc (`Use rpk -X flag to override any rpk-specific configuration option.`)? If so, I think it would be good to make that connection for the reader, perhaps by expanding the first line a bit, e.g.:
> ...rpk-specific configuration option set by default in `redpanda.yaml`

(something along those lines)

Reviewer 2: edit this on the opening paragraph 

```
`rpk` resolves configuration values in the following priority order where higher priority options override lower priority ones:
```

|<<registry-tls-enabled,registry.tls.enabled>> |RPK_REGISTRY_TLS_ENABLED
|<<registry-tls-insecure-skip-verify,registry.tls.insecure_skip_verify>> |RPK_REGISTRY_TLS_INSECURE_SKIP_VERIFY
|<<registry-tls-ca,registry.tls.ca>> |RPK_REGISTRY_TLS_CA
|<<registry-tls-cert,registry.tls.cert>> |RPK_REGISTRY_TLS_CERT
|<<registry-tls-key,registry.tls.key>> |RPK_REGISTRY_TLS_KEY
|<<cloud-client-id,cloud.client_id>> |RPK_CLOUD_CLIENT_ID
|<<cloud-client-secret,cloud.client_secret>> |RPK_CLOUD_CLIENT_SECRET
|<<globals-prompt,globals.prompt>> |RPK_GLOBALS_PROMPT
|<<globals-no-default-cluster,globals.no_default_cluster>> |RPK_GLOBALS_NO_DEFAULT_CLUSTER
|<<globals-command-timeout,globals.command_timeout>> |RPK_GLOBALS_COMMAND_TIMEOUT
|<<globals-dial-timeout,globals.dial_timeout>> |RPK_GLOBALS_DIAL_TIMEOUT
|<<globals-request-timeout-overhead,globals.request_timeout_overhead>> |RPK_GLOBALS_REQUEST_TIMEOUT_OVERHEAD
|<<globals-retry-timeout,globals.retry_timeout>> |RPK_GLOBALS_RETRY_TIMEOUT
|<<globals-fetch-max-wait,globals.fetch_max_wait>> |RPK_GLOBALS_FETCH_MAX_WAIT
|<<globals-kafka-protocol-request-client-id,globals.kafka_protocol_request_client_id>> |RPK_GLOBALS_KAFKA_PROTOCOL_REQUEST_CLIENT_ID
|===

> == Quick configuration examples

Reviewer 3: ```suggestion
== Example configurations
```
To make it a bit more concise. Or "Configuration examples"?

Reviewer 2: configuration examples sound better. I'll change

Reviewer 2: ```suggestion
== Configuration examples
```

> For regular use, create an xref:get-started:config-rpk-profile.adoc[rpk profile]:

Reviewer 3: ```suggestion
* To persist configuration across sessions, use an xref:get-started:config-rpk-profile.adoc[rpk profile]:
+
```
"For regular use" is a bit vague

I'd also group the related commands together, for example have "Create a profile" and "Use the profile" go together with one list item "For regular use..." Then "For temporary use" would be the next list item, and "Run commands that use env vars..." is grouped with it. I'll try to add those in as suggestions that are committable

Reviewer 2: Removed for reglar use.

I tried using the grouping but it didn't look quite as good. Besides they're supposed to be some examples so it doesn't make much sense to group them. Removed it.

> Create a persistent profile:

Reviewer 3: ```suggestion
Create a profile:
```

> 

Reviewer 3: ```suggestion
+
```

```
rpk profile create <profile-name> \
  --set brokers=<broker1:port>,<broker2:port> \
  --set user=<username> \
  --set pass=<password> \
  --set sasl.mechanism=<mechanism> \
  --set tls.enabled=true \
  --description "<profile-description>"
```

> 

Reviewer 3: ```suggestion
+
```

Use the profile for commands:

> 

Reviewer 3: ```suggestion
+
```

```
rpk topic list --profile <profile-name>
```

> For temporary use or automation scripts, set environment variables:

Reviewer 3: ```suggestion
* For temporary use or automation scripts, set environment variables:
```

> 

Reviewer 3: ```suggestion
+
```

```
export RPK_BROKERS="<broker1:port>,<broker2:port>"
export RPK_USER="<username>"
export RPK_PASS="<password>"
export RPK_SASL_MECHANISM="<mechanism>"
export RPK_TLS_ENABLED="true"
```

> 

Reviewer 3: ```suggestion
+
```

> Run commands that use environment variables automatically:

Reviewer 3: I had no idea about this, do we have a definitive list of such commands? I'm not sure that we indicate this on the respective rpk command reference pages.

Reviewer: I think here should be just `Run a command`.

All the configuration properties are loaded by default in all the commands.

> 

Reviewer 3: ```suggestion
+
```

```
rpk topic list
```

== Options

=== Duration format

Duration values use Go's standard duration format. A duration string is a sequence of decimal numbers with unit suffixes:

- `ns` = nanoseconds
- `us` or `µs` = microseconds  
- `ms` = milliseconds
- `s` = seconds
- `m` = minutes
- `h` = hours

*Examples*: `30s`, `1m30s`, `2h`, `500ms`, `1h15m30s`

You can combine multiple units: `2h45m30s` means 2 hours, 45 minutes, and 30 seconds.

> The following options are available:

Reviewer 3: I think we should make a clearer delineation for Duration format vs the options; as this stands, it sounds like "The following options" still belongs to the "Duration format" section. I might actually move "Duration format" to a different section that's suitable, or maybe take it out entirely and have it as the same level as "Options" (my suggestion would be after "Environment variables" and before "Configuration examples")

=== brokers

> A comma-delimited list of broker `host:port`'s to connect to the Kafka API.

Reviewer 3: ```suggestion
A comma-delimited list of broker `host:port` pairs to connect to the Kafka API.
```
The formatting for `host:port` didn't render correctly, my guess is it has to do with the apostrophe immediately following it?

== Configuration priority

`rpk` resolves configuration values in the following priority order (highest to lowest):

1. **Command-line flags** (including `-X` options): Highest priority, applies to current command only
2. **Environment variables**: Medium priority, lasts for shell session
   - **RPK_*** variables (modern): Higher precedence
   - **REDPANDA_*** variables (legacy): Lower precedence
3. **rpk profile** (`rpk.yaml`): Lower priority, persistent across sessions (recommended)
4. **Redpanda configuration** (`redpanda.yaml` rpk section): Lowest priority, system-wide defaults

> NOTE: If both `RPK_*` and `REDPANDA_*` environment variables are set for the same option, the `RPK_*` variable takes precedence.

Reviewer 2: @r-vasquez can you confirm this info? I added because of the code review but I'm not sure.

Reviewer: I believe this is correct, but we should remove trace of the old REDPANDA_ and only document the RPK_ ones moving forward :+1:

=== Legacy REDPANDA_* environment variables

> For backward compatibility, `rpk` supports these specific legacy environment variables. Note that not all `-X` options have `REDPANDA_*` equivalents - only certain commonly used connection and authentication options:

Reviewer 2: @r-vasquez during the code review, I noticed that REDPANDA_TLS_ENABLED is not present in https://github.com/redpanda-data/redpanda/blob/dev/src/go/rpk/pkg/config/params.go#L1569

Not sure if intentional, that's why I added this note.

Reviewer: Yeah, the new ones are not 1:1 with the old ones. We had some variables scattered around and we still support them for backwards compatibility.
